src/client/scripts/esm/chess/variants/variant.ts

```diff
@@ -1,13 +1,24 @@
 
+// @ts-ignore
```

Comment:
All these repeated ts-ignore lines are kind of ugly and probably not sustainable for every single file. What's the issue we get without them?

Reply:
The ts-ignore lines are required since the .js files don't have corresponding .d.ts files; they should be removable once everything is converted though.

Reply:
What if we tried adding these to tsconfig?
```js
    "allowJs": true,        // Allows JavaScript files to be imported
    "checkJs": false,       // Disables type checking for JavaScript files
    "esModuleInterop": true, // Enables compatibility with ES module syntax
    "skipLibCheck": true    // Skips type-checking for declaration files
```